client/utils/profile-obj-compress.js

```diff
@@ -0,0 +1,83 @@
+export const compressProfile = (profile) => {
+
+  const {_id, photo} = profile;
+  const { age, fName, lName, occupation } = profile.personal;
+  const { country, place } = profile.personal.currentLocation;
+  const { cv, github, linkedIn, website } = profile.networking;
+
+  const compressed = { _id, cv, github, linkedIn, website, age, country, place, fName, lName, occupation, photo};
+  return compressed;
+};
+
+let emptyProfile = {
+  "_id": null,
+  "contacts": [],
+  "networking": {
+    "cv": "",
+    "github": "",
+    "linkedIn": "",
+    "website": "",
+  },
+  "personal": {
+    "age": 99,
+    "birthday": "",
+    "birthplace": {
+      "country": "",
+      "lat": 0,
+      "lng": 0,
+      "place": "",
+    },
+    "currentLocation": {
+      "country": "",
+      "lat": 0,
+      "lng": 0,
+      "place": "",
+    },
+    "fName": "",
+    "lName": "",
+    "nationality": "",
+    "occupation": "",
+  },
+  "photo": "",
+  "social": {
+    "blog": "",
+    "facebook": "",
+    "instagram": "",
+    "twitter": "",
+  }
+}
+
+export const decompressProfile = (compressedProfile) => {
+  let decompressed = Object.assign({}, emptyProfile);
+
+  const { cv, github, linkedIn, website, age, country, place, fName, lName, occupation} = compressedProfile;
+
+  let { _id, photo} = compressedProfile;
+  let currentLocation = Object.assign({}, { country, place });
+  let personal = Object.assign({}, { age, currentLocation, fName, lName, occupation});
+  let networking = Object.assign({}, {cv, github, linkedIn, website});
```

Comment:
The law says 10 push-ups per `let` that should be `const`. You're gonna get ripped as Arnold.

![](https://media.giphy.com/media/q57n5FWtPcFOg/giphy.gif)